automata/regex/parser.py

```diff
@@ -181,9 +181,10 @@ def concatenate(self, other):
 
         self._final_states = other._final_states
 
-    def kleene_star(self):
+    def kleene(self, *, is_kleene_star):
```

Comment:
@eliotwrobson Since it looks like the only thing that separates the Kleene Star operation from Kleene Plus is an extra conditional at the end, i.e.:

```python
if is_kleene_star:
    self._final_states.add(new_initial_state)
```

Given this, I almost feel it would be better to rename your `kleene()` method to `kleene_plus` without the `is_kleene_star` keyword argument, _then_ define `kleene_star` as a small wrapper around `kleene_plus` like so:

```python
def kleene_star(self):
    self.kleene_plus()
    self._final_states.add(self._initial_state)
```

If I wrote this correctly, I think it still checks out logically—just now that we can eliminate the `kleene()` middleman and just have separate `kleene_plus()` and `kleene_star()` methods.

Thoughts?

Reply:
I think I slightly prefer the way I have it to lower the number of different methods, but your way is probably easier to understand for someone unfamiliar reading the code. Will change.